packages/manager-react-components/src/hooks/datagrid/useIcebergV6.tsx

```diff
@@ -35,24 +35,26 @@ export function useResourcesIcebergV6<T = unknown>({
         sortReverse: sorting?.desc,
       }),
     getNextPageParam: (lastPage, _allPages, lastPageIndex) => {
-      if (lastPage.totalCount / pageSize > lastPageIndex) return null;
-      return lastPageIndex + 1;
+      if (lastPage.totalCount / pageSize > lastPageIndex) {
+        return lastPageIndex + 1;
+      }
+      return null;
     },
     select: (data) => {
       const pageIndex = data.pageParams[data.pageParams.length - 1];
       const { totalCount } = data.pages[0];
+
       return {
         data,
         pageIndex,
         totalCount,
         flattenData: data.pages.flatMap((page) => page.data),
-        hasNextPage: totalCount / pageSize > pageIndex,
       };
     },
   });
 
   return {
-    ...(dataSelected ?? { ...dataSelected, totalCount: 0, hasNextPage: false }),
+    ...(dataSelected ?? { ...dataSelected, totalCount: 0 }),
```

Comment:
if `dataSelected` is null or undefined the spread operator is useless `{ ...dataSelected }`

Reply:
It's for Typescript. If I remove It. I have a build error because `flattenData` not exist. So I can't decompose it by example

Reply:
But I would have made a mistake when I decomposed. So I can assume that it's always defined